client/src/resources/asset.rs

```diff
@@ -253,6 +255,97 @@ mod asset_identifier_tests {
     }
 }
 
+/// When a bad token or string is provided to parsing into an asset
+/// you get an error.
+#[derive(Debug)]
+pub struct ParseAssetError {
+    details: String,
+}
+
+impl ParseAssetError {
+    /// Create from a string reference
+    pub fn from_str(details: &str) -> ParseAssetError {
+        ParseAssetError {
+            details: details.to_string(),
+        }
+    }
+
+    /// Public method for getting the details of the error
+    pub fn description(&self) -> &str {
+        &self.details
+    }
+}
+
+impl fmt::Display for ParseAssetError {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "{}", self.details)
+    }
+}
+
+impl FromStr for AssetIdentifier {
```

Comment:
I see we are adding this trait impl here but I don't see it removed from the CLI, is there a reason for that?

Reply:
Will hold off for a future PR -- I should probably create a new issue